File: csharp/DVSClient/DVSClient/address/search/Result.cs
using DVSClient.Common;
using DVSClient.Server.Address.Search;

namespace DVSClient.Address.Search
{
    public class Result
    {
        public bool MoreResultsAvailable { get; }
        public Confidence? Confidence { get; }
        public string SuggestionsKey { get; }
        public string SuggestionsPrompt { get; }
        public IEnumerable<Suggestion> Suggestions { get; } = new List<Suggestion>();

        public Result(RestApiAddressSearchResponse response)
        {
            var apiResult = response.Result;
            if (apiResult != null)
            {
                MoreResultsAvailable = apiResult.IsMoreResultsAvailable() ?? false;
                Confidence = apiResult.Confidence?.GetEnumValueFromJsonName<Confidence>();
                SuggestionsKey = apiResult.SuggestionsKey ?? string.Empty;
                SuggestionsPrompt = apiResult.SuggestionsPrompt ?? string.Empty;
                Suggestions = apiResult.Suggestions?.Select(s => new Suggestion(s)).ToList() ?? new List<Suggestion>();
            }
            else
            {
                MoreResultsAvailable = false;
                Confidence = null;
                SuggestionsKey = string.Empty;
                SuggestionsPrompt = string.Empty;
                Suggestions = new List<Suggestion>();
            }
        }
    }
}

File: java/src/main/java/com/experian/dvs/client/address/search/Result.java
package com.experian.dvs.client.address.search;

import com.experian.dvs.client.address.Confidence;
import com.experian.dvs.client.server.address.search.RestApiAddressSearchResponse;
import com.experian.dvs.client.server.address.search.RestApiAddressSearchResult;

import java.util.List;
import java.util.Objects;

public class Result {

    private final boolean moreResultsAvailable;
    private final Confidence confidence;
    private final String suggestionsKey;
    private final String suggestionsPrompt;
    private final List<Suggestion> suggestions;


    public Result(final RestApiAddressSearchResponse response) {
        final RestApiAddressSearchResult apiResult = response.getResult();

        if (apiResult != null) {
            this.moreResultsAvailable = Boolean.TRUE.equals(apiResult.isMoreResultsAvailable());
            this.confidence = apiResult.getConfidence() != null ? Confidence.fromName(apiResult.getConfidence()) : null;
            this.suggestionsKey = Objects.toString(apiResult.getSuggestionsKey(), "");
            this.suggestionsPrompt = Objects.toString(apiResult.getSuggestionsPrompt(), "");
            this.suggestions = apiResult.getSuggestions() != null ? apiResult.getSuggestions().stream().map(Suggestion::new).toList() : List.of();
        } else {
            this.moreResultsAvailable = false;
            this.confidence = null;
            this.suggestionsKey = "";
            this.suggestionsPrompt = "";
            this.suggestions = List.of();
        }
    }

    public boolean isMoreResultsAvailable() {
        return moreResultsAvailable;
    }

    public Confidence getConfidence() {
        return confidence;
    }

    public String getSuggestionsKey() {
        return suggestionsKey;
    }

    public String getSuggestionsPrompt() {
        return suggestionsPrompt;
    }

    public List<Suggestion> getSuggestions() {
        return suggestions;
    }
}
